### PR TITLE
hunt down DAML

### LINKED ISSUE #13190: Fix remaining incorrect spelling of Daml
There are some place where we still have Daml spelled as an acronym. As the name no longer stands for anything, it would be nice to remove those. Some care should be taken as some occurrences are valid (e.g. in environment variables) so you cannot really do a batch re-write of all occurrences.

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
Process:

- `git ls-files | xargs sed -i 's/DAML/Daml/g'`
- `git add -p`

To review:

- Check for false positives by carefully reviewing the diff in this PR.
- Check for false negatives with `git grep DAML`.

Almost fixes #13190

CHANGELOG_BEGIN

- Removed all traces of DAML. This may affect error messages and debug
  outputs.

CHANGELOG_END

### CODE REVIEW

File: compiler/daml-extension/package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
                     ".daml-core"
                 ],
                 "aliases": [
-                    "DAML CORE"
+                    "Daml CORE"
```
Comment (realvictorprm) [inline compiler/daml-extension/package.json:39]: Probably core should be writen lowercase too then

Comment (realvictorprm) [inline compiler/daml-extension/package.json:39]:
```suggestion
                    "Daml Core"
```

File: compiler/damlc/daml-stdlib-src/DA/Text.daml [hunk 2]
```diff
@@ -325,7 +325,7 @@ sha256 = primitive @"BESha256Text"
 
 -- | Reverse some `Text`.
 -- ```
---   reverse "DAML" == "LMAD"
+--   reverse "Daml" == "lmaD"
```
Comment (realvictorprm) [inline compiler/damlc/daml-stdlib-src/DA/Text.daml:328]: ok this one is really funny lol

**Review by realvictorprm — APPROVED** [review]
Is nice! Thank you! :)

**Comment by realvictorprm** [comment]
Side note, I didn't check that you removed all occurrences but I checked that all what _is_ changed is correct.

**Comment by garyverhaegen-da** [comment]
> Side note, I didn't check that you removed all occurrences

I did and I don't. The given `sed` command fails on a file with a quote in it, so this only affected the first 200 files or so.

**Review by cocreature — APPROVED** [review]
lgtm thanks

**Pull request merged**